src/mapping.ts

```diff
@@ -92,16 +126,50 @@ function deregisterPodLogic(id: string): void {
     oldAdmin.save();
   }
 
-  // Delete the Pod
-  store.remove('Pod', id);
-}
+  // Delete the PodUsers for this Pod
+  if (pod.users.length > 0) {
+    for (let i = 0; i < pod.users.length; i++) {
+      // Find the user
+      let podUserId = pod.users[i].toString();
 
-export function handleDeregisterPodV1_3(event: DeregisterPodV1_3): void {
-  let id = event.params.podId.toString();
-  deregisterPodLogic(id);
+      // Ex: "0x1...-123" -> "0x1..."
+      let userId = podUserId.substring(0, id.length + 1);
```

Comment:
🤔 I'd rather the mapping file not need to know these ID details, but I don't know a better way to fetch the user's id

Reply:
feel like thats the only way to get it right? how else would you want to handle this